tests/data/model-validation.ts

```diff
@@ -135,4 +135,647 @@ type org
+        relatedInformation: {
+          relation: "allowed",
+          type: "missing-definition",
+        },
+        severity: 8,
+        source: "linter",
+        startColumn: 32,
+        startLineNumber: 5,
+      },
+    ],
+  },
+  {
+    name: "model 1.1 base in exclusion not valid",
+    friendly: `type user
+type group
+  relations
+    define member: [user] as self
+    define reader as allowed but not member
+`,
+    expectedError: [
+      {
+        endColumn: 29,
+        endLineNumber: 5,
+        message: "The relation `allowed` does not exist.",
+        relatedInformation: {
+          relation: "allowed",
+          type: "missing-definition",
+        },
+        severity: 8,
+        source: "linter",
+        startColumn: 22,
+        startLineNumber: 5,
+      },
+    ],
+  },
+  {
+    name: "model 1.1 diff in exclusion not valid",
+    friendly: `type user
+type group
+  relations
+    define member: [user] as self
+    define reader as member but not allowed
+`,
+    expectedError: [
+      {
+        endColumn: 44,
+        endLineNumber: 5,
+        message: "The relation `allowed` does not exist.",
+        relatedInformation: {
+          relation: "allowed",
+          type: "missing-definition",
+        },
+        severity: 8,
+        source: "linter",
+        startColumn: 37,
+        startLineNumber: 5,
+      },
+    ],
+  },
+  {
+    name: "model 1.1 diff in exclusion not valid and spaces are reflected correctly in error messages",
+    friendly: `type user
+type group
+  relations
+    define member: [user] as self
+    define reader   as member   but not   allowed
+`,
+    expectedError: [
+      {
+        endColumn: 50,
+        endLineNumber: 5,
+        message: "The relation `allowed` does not exist.",
+        relatedInformation: {
+          relation: "allowed",
+          type: "missing-definition",
+        },
+        severity: 8,
+        source: "linter",
+        startColumn: 43,
+        startLineNumber: 5,
+      },
+    ],
+  },
+  {
+    name: "empty directly assignable relations with spaces should yield error",
+    friendly: `type user
+type org
+  relations
+    define member: [ ] as self
+    define reader: [user] as self
+`,
+    expectedError: [
+      {
+        endColumn: 18,
+        endLineNumber: 4,
+        message: "`member` is an impossible relation (no entrypoint).",
+        relatedInformation: {
+          relation: "member",
+          type: "relation-no-entry-point",
+        },
+        severity: 8,
+        source: "linter",
+        startColumn: 12,
+        startLineNumber: 4,
+      },
+    ],
+  },
+  {
+    name: "empty directly assignable relations without spaces should yield error",
+    friendly: `type user
+type org
+  relations
+    define member: [] as self
+    define reader: [user] as self
+`,
+    expectedError: [
+      {
+        endColumn: 18,
+        endLineNumber: 4,
+        message: "`member` is an impossible relation (no entrypoint).",
```

Comment:
This looks weird.. can we change it to `member needs to define the allowable types`?

Reply:
Let's defer updating the exact message wording to a later PR when there are consensus.